### PR TITLE
1652 remove term assignment feature

### PR DESCRIPTION
**Pull request by ledsoft**
https://kbss.felk.cvut.cz/redmine/issues/1652

### CODE REVIEW

File: src/main/java/cz/cvut/kbss/termit/dto/occurrence/ResourceTermOccurrences.java [hunk 1]
```diff
@@ -45,7 +43,13 @@
                 @VariableResult(name = "suggested", type = Boolean.class)
         }
 ))
-public class ResourceTermOccurrences extends ResourceTermAssignments {
+public class ResourceTermOccurrences extends AbstractTermOccurrenceInfo {
```
Comment (psiotwo) [inline src/main/java/cz/cvut/kbss/termit/dto/occurrence/ResourceTermOccurrences.java]: OK. Yet, such changes will make any future continuation on the closed-source TermIt even less mergeable with the open-source. However, probably not a big deal anyway.

File: src/main/java/cz/cvut/kbss/termit/service/repository/TermOccurrenceRepositoryService.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    @Override
+    public TermOccurrence getRequiredReference(URI id) {
+        return termOccurrenceDao.getReference(id).orElseThrow(() ->
+                NotFoundException.create(TermOccurrence.class.getSimpleName(), id)
+        );
+    }
+
+    @Transactional
+    @Override
+    public void persistOccurrence(TermOccurrence occurrence) {
```
Comment (psiotwo) [inline src/main/java/cz/cvut/kbss/termit/service/repository/TermOccurrenceRepositoryService.java]: Why not align service names with other services.

**Review by psiotwo — DISMISSED** [review]
lgtm

File: src/main/java/cz/cvut/kbss/termit/dto/occurrence/ResourceTermOccurrences.java (hunk 1 above)
Comment (ledsoft) [reply src/main/java/cz/cvut/kbss/termit/dto/occurrence/ResourceTermOccurrences.java]: So you think the package/class renaming should be reverted?

Comment (psiotwo) [reply src/main/java/cz/cvut/kbss/termit/dto/occurrence/ResourceTermOccurrences.java]: Yes, I think keeping the two versions as close as possible until there is a good reason not to do it (e.g. breaking change in the code) is a better option now.

Comment (ledsoft) [reply src/main/java/cz/cvut/kbss/termit/dto/occurrence/ResourceTermOccurrences.java]: Ok, good point. Will do.

File: src/main/java/cz/cvut/kbss/termit/dto/assignment/AbstractAssignmentInfo.java [hunk 3]
```diff
@@ -26,7 +26,7 @@
 import java.util.Objects;
 import java.util.Set;
 
-abstract class AbstractAssignmentsInfo implements HasTypes {
+abstract class AbstractAssignmentInfo implements HasTypes {
```
Comment (psiotwo) [inline src/main/java/cz/cvut/kbss/termit/dto/assignment/AbstractAssignmentInfo.java]: it was AbstractAssignment**s**Info

**Pull request merged**